lib/drives/pages/drive_chat_page.dart

```diff
@@ -16,107 +18,140 @@ class DriveChatPage extends StatefulWidget {
+          ? StreamBuilder<List<Message>>(
+              stream: _messagesStream,
+              builder: (BuildContext context, AsyncSnapshot<List<Message>> snapshot) {
+                if (snapshot.hasData) {
+                  for (final Message message in snapshot.data!) {
+                    final Chat chat = _chats.firstWhere((Chat chat) => chat.id == message.chatId);
+                    if (chat.messages!.contains(message)) {
+                      chat.messages!.remove(message);
+                    }
+                    chat.messages!.add(message);
+                  }
+                  final List<Card> widgets = _chats.map<Card>((Chat chat) => _buildChatWidget(chat)).toList();
+                  return ListView.separated(
+                    itemCount: widgets.length,
+                    itemBuilder: (BuildContext context, int index) {
+                      return widgets[index];
+                    },
+                    separatorBuilder: (BuildContext context, int index) {
+                      return const SizedBox(height: 10);
+                    },
+                  );
+                } else {
+                  return const Center(
+                    child: CircularProgressIndicator(),
+                  );
+                }
+              },
+            )
+          : Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: <Widget>[
+                Image.asset('assets/chat_shrug.png', scale: 8),
+                const SizedBox(height: 16),
+                Text(
+                  S.of(context).pageChatEmptyTitle,
+                  style: Theme.of(context).textTheme.headline6,
+                ),
+                const SizedBox(height: 8),
+                Padding(
+                  padding: const EdgeInsets.symmetric(horizontal: 16),
+                  child: Text(
+                    S.of(context).pageDriveChatEmptyMessage,
+                    textAlign: TextAlign.center,
+                  ),
+                ),
+              ],
+            ),
     );
   }
 
-  Widget _riderList(Set<Profile> riders) {
-    Widget ridersColumn = Container();
-    if (riders.isNotEmpty) {
-      ridersColumn = Column(
-        children: List<Padding>.generate(
-          riders.length,
-          (int index) => Padding(
-            padding: const EdgeInsets.symmetric(horizontal: 5.0, vertical: 5.0),
-            child: ProfileWidget(
-              riders.elementAt(index),
-              //TODO go to chat
-              onTap: () {},
-              actionWidget: const Icon(
-                Icons.chat,
-                color: Colors.black,
-                size: 36.0,
+  Card _buildChatWidget(Chat chat) {
+    chat.messages!.sort((Message a, Message b) => b.createdAt!.compareTo(a.createdAt!));
+    final Message? lastMessage = chat.messages!.isEmpty ? null : chat.messages!.first;
+    final Widget? subtitle = lastMessage == null
+        ? null
+        : Wrap(
```

Comment:
this should also not be shown if the last message was not frome me. That indicator just does not make sense in that case

Reply:
yes thanks